subsys/bluetooth/shell/has.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file
+ * @brief Bluetooth Hearing Access Service (HAS) shell.
+ *
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/audio/has.h>
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "bt.h"
+
+static int cmd_preset_reg(const struct shell *sh, size_t argc, char **argv)
+{
+	int err;
+	struct bt_has_preset_register_param param = {
+		.index = strtol(argv[1], NULL, 0),
```

Comment:
Consider constraining the base to either decimal or hex if there is a convention. It avoids the issue of a leading 0 setting the base to 8.